lib/distribute_reads/global_methods.rb

```diff
@@ -20,8 +20,11 @@ def distribute_reads(**options)
         max_lag = options[:max_lag]
         if max_lag && !options[:primary]
           Array(options[:lag_on] || [ActiveRecord::Base]).each do |base_model|
-            if DistributeReads.lag(connection: base_model.connection) > max_lag
+            if DistributeReads.lag(connection: base_model.connection) > max_lag || DistributeReads.lag(connection: base_model.connection) == -1
```

Comment:
nit: Maybe use something like `current_lag = DistributeReads.lag(connection: base_model.connection)` to simplify the code. Line is getting longer.